llvm/include/llvm/IR/IntrinsicsEVM.td

```diff
@@ -118,35 +119,43 @@ def int_evm_origin : Intrinsic<[llvm_i256_ty], [], [IntrNoMem, IntrWillReturn]>;
 def int_evm_caller : Intrinsic<[llvm_i256_ty], [], [IntrNoMem, IntrWillReturn]>;
 
 def int_evm_balance : Intrinsic<[llvm_i256_ty], [llvm_i256_ty],
-                                [IntrReadMem, IntrInaccessibleMemOnly, IntrWillReturn]>;
+                                [IntrReadMem, IntrInaccessibleMemOnly, IntrNoCallback,
+                                 IntrWillReturn]>;
 
 def int_evm_callvalue : Intrinsic<[llvm_i256_ty], [], [IntrNoMem, IntrWillReturn]>;
 
 def int_evm_calldatasize : Intrinsic<[llvm_i256_ty], [], [IntrNoMem, IntrWillReturn]>;
 
 def int_evm_calldataload
   : Intrinsic<[llvm_i256_ty], [LLVMQualPointerType<AS.CALL_DATA>],
-              [IntrReadMem, IntrArgMemOnly]>;
+              [IntrReadMem, IntrArgMemOnly, IntrNoCallback]>;
 
 def int_evm_codesize : Intrinsic<[llvm_i256_ty], [], [IntrNoMem, IntrWillReturn]>;
 
 def int_evm_gasprice : Intrinsic<[llvm_i256_ty], [], [IntrNoMem, IntrWillReturn]>;
 
+// This should have 'read of inaccessible' memory along with read/write memory
```

Comment:
Could you clarify this documentation? Is it something left to do? Why can't we do it with this patch?